sdks/go/pkg/beam/core/runtime/exec/pardo.go

```diff
@@ -95,26 +96,56 @@ func (n *ParDo) ProcessElement(ctx context.Context, elm FullValue, values ...ReS
 	}
 
 	ctx = metrics.SetPTransformID(ctx, n.PID)
+	fn := n.Fn.ProcessElementFn()
 
-	val, err := n.invokeDataFn(ctx, elm.Timestamp, n.Fn.ProcessElementFn(), &MainInput{Key: elm, Values: values})
-	if err != nil {
-		return n.fail(err)
-	}
+	// If the function observes windows, we must pass invoke it for each window. The expected fast path
```

Comment:
Grammar in first sentence. Perhaps, ", we must invoke it for each window?"

Reply:
Done